packages/lit-html/rollup.config.js

```diff
@@ -53,82 +54,129 @@ const entryPoints = [
   'parts',
 ];
 
-// Shared name cache
-// Ideally we'd share this only between lit-html.js and parts.js and force
-// directives to use the public APIs.
+// The Terser shared name cache allows us to mangle the names of properties
+// consistently across modules, so that e.g. parts.js can safely access internal
+// details of lit-html.js.
+//
+// However, we still have to account for the problem of mangled names getting
+// re-used for different properties across files, because Terser does not
+// consult the nameCache to decide whether a mangled name is available or not.
+//
+// For example:
+//
+// file1:
+//   obj.foo -> A
+//   obj.bar -> B
+//
+// file2:
+//   obj.bar -> B (Correctly chosen from nameCache.)
+//   obj.baz -> A (Oops, foo and baz are different properties on the same
+//                 object, but now they both have the same mangled name,
+//                 which could result in very unpredictable behavior).
+//
+// To trick Terser into doing what we need here, we first create a giant bundle
+// of all our code in a single file, tell Terser to minify that, and then throw
+// it away. This seeds the name cache in a way that guarantees every property
+// gets a unique mangled name.
 const nameCache = {};
+const nameCacheSeederInfile = 'name-cache-seeder-virtual-input.js';
+const nameCacheSeederOutfile = 'name-cache-seeder-throwaway-output.js';
+const nameCacheSeederContents = entryPoints
```

Comment:
The original idea was that we would make our directives use the public API and only share names between lit-html.ts and parts.ts. Accessing private APIs from directives probably means that our public API is deficient.

We can do that here by just including those two files, right?

Reply:
We could do that (though not just by changing this array -- we'd need to make separate batches of rollup runs, because this array is only related to the "don't mangle two different properties in two different files to the same name" concern, not the "do mangle the same property in two different files to the same name" concern).

Currently, `directives/live.ts` and `directives/repeat.ts` both directly write to `NodePart._value` to bypass dirty checking, so before we did this, we would need to resolve what that public interface should be. A `setPartValueDirectly` or something function? Looks like https://github.com/Polymer/lit-html/issues/1261 is the place for this discussion.